### ISSUE TITLE
`currentOptions` uninitialized (Vite)

### ISSUE THREAD

**Issue by devlsh** [Akryum/vue-apollo#1213]
**Describe the bug**
I'm using Vite and `@vue/apollo-composable`@`4.0.0-alpha.13`, and the browser is printing this error:

![](https://i.imgur.com/VlOdB5V.png)

(See https://github.com/vuejs/vue-apollo/blob/v4/packages/vue-apollo-composable/src/useQuery.ts#L358)

**To Reproduce**

**Expected behavior**

**Versions**
vue: `next`
vue-apollo: `4.0.0-alpha.13`
apollo-client: `3.3.20`

**Comment by Asim-Tahir** [comment]
The temporal solution is downgrade to vue-apollo `4.0.0-alpha.12`

**Comment by maartenvn** [comment]
Same error when using `@nuxtjs/apollo 4.0.1-rc.5` and `@nuxtjs/composition-api 0.24.6`

**Comment by kenneth-hendricks** [comment]
Also having this issue with `@nuxtjs/apollo 4.0.1-rc.5`  and `@nuxtjs/composition-api 0.24.6`, though it does work client side, this error only occurs during SSR

**Comment by exreplay** [comment]
Thx to this PR #1218 im currently using a `postinstall` script which moves `currentOptions` up to the correct position.

_scripts/vue-apollo-patch.js_
```js
const fs = require('fs');
const path = require('path');

const indexEsmPath = path.resolve(
  __dirname,
  '../node_modules/@vue/apollo-composable/dist/index.esm.js'
);

fs.writeFileSync(
  indexEsmPath,
  fs
    .readFileSync(indexEsmPath, 'utf8')
    .replace(/const currentOptions = ref3\(\);/, '')
    .replace(
      /const documentRef = paramToRef\(document\);/,
      'const currentOptions = ref3();\nconst documentRef = paramToRef(document);'
    )
);

const indexPath = path.resolve(
  __dirname,
  '../node_modules/@vue/apollo-composable/dist/index.js'
);

fs.writeFileSync(
  indexPath,
  fs
    .readFileSync(indexPath, 'utf8')
    .replace(/const currentOptions = \(0, import_vue_demi8\.ref\)\(\);/, '')
    .replace(
      /const documentRef = paramToRef\(document\);/,
      'const currentOptions = (0, import_vue_demi8.ref)();\nconst documentRef = paramToRef(document);'
    )
);
```

_package.json_
```json
{
    "scripts": {
        "postinstall": "node scripts/vue-apollo-patch.js"
    }
}
```

Hope this helps.

**Comment by exreplay** [comment]
I updated the script which should move all `currentOptions` to the correct position.

```js
const fs = require('fs');
const path = require('path');

const indexEsmPath = path.resolve(
  __dirname,
  '../node_modules/@vue/apollo-composable/dist/index.esm.js'
);

fs.writeFileSync(
  indexEsmPath,
  fs
    .readFileSync(indexEsmPath, 'utf8')
    .replace(/const currentOptions = ref3\(\);/, '')
    .replace(
      /const vm = getCurrentInstance3\(\);/,
      'const vm = getCurrentInstance3();\nconst currentOptions = ref3();'
    )
    .replace(/const currentOptions = ref5\(\);/, '')
    .replace(
      /const vm = getCurrentInstance5\(\);/,
      'const vm = getCurrentInstance5();\nconst currentOptions = ref5();'
    )
);

const indexPath = path.resolve(
  __dirname,
  '../node_modules/@vue/apollo-composable/dist/index.js'
);

fs.writeFileSync(
  indexPath,
  fs
    .readFileSync(indexPath, 'utf8')
    .replace(/const currentOptions = \(0, import_vue_demi8\.ref\)\(\);/, '')
    .replace(
      /const vm = \(0, import_vue_demi8\.getCurrentInstance\)\(\);/,
      'const vm = (0, import_vue_demi8.getCurrentInstance)();\nconst currentOptions = (0, import_vue_demi8.ref)();'
    )
    .replace(/const currentOptions = \(0, import_vue_demi5\.ref\)\(\);/, '')
    .replace(
      /const vm = \(0, import_vue_demi5\.getCurrentInstance\)\(\);/,
      'const vm = (0, import_vue_demi5.getCurrentInstance)();\nconst currentOptions = (0, import_vue_demi5.ref)();'
    )
);
```

**Comment by sinisterstumble** [comment]
I see the same thing

**Comment by xkisu** [comment]
Can confirm it was just fixed in `4.0.0-alpha.14` 👍
